tests/test_generator.py

```diff
@@ -0,0 +1,1202 @@
+        [
+            mocker.call("Labels to be fetched from: %s.", config_repository.query_labels),
+            mocker.call("Fetching issues with label `%s`.", "bug"),
+            mocker.call("Fetching issues with label `%s`.", "enhancement"),
+        ],
+        any_order=True,
+    )
+
+
+def test_fetch_github_issues_repository_none(mocker, generator, config_repository):
+    # Arrange
+    mock_get_repo = generator._LivingDocumentationGenerator__github_instance.get_repo
+    mock_get_repo.return_value = None
+    mocker.patch(
+        "living_documentation_generator.generator.ActionInputs.get_repositories", return_value=[config_repository]
+    )
+
+    # Act
+    actual = generator._fetch_github_issues()
+
+    # Assert
+    assert {} == actual
+    mock_get_repo.assert_called_once_with("test_org/test_repo")
+
+
+# _fetch_github_project_issues
+
+
+def test_fetch_github_project_issues_correct_behaviour(mocker, generator):
+    # Arrange
+    mocker.patch(
+        "living_documentation_generator.generator.ActionInputs.get_is_project_state_mining_enabled", return_value=True
+    )
+    mock_logger_info = mocker.patch("living_documentation_generator.generator.logger.info")
+    mock_logger_debug = mocker.patch("living_documentation_generator.generator.logger.debug")
+
+    repository_1 = mocker.Mock()
+    repository_1.organization_name = "OrgA"
+    repository_1.repository_name = "RepoA"
+    repository_1.projects_title_filter = ""
+
+    repository_2 = mocker.Mock()
+    repository_2.organization_name = "OrgA"
+    repository_2.repository_name = "RepoB"
+    repository_2.projects_title_filter = "ProjectB"
+
+    mocker.patch(
+        "living_documentation_generator.generator.ActionInputs.get_repositories",
+        return_value=[repository_1, repository_2],
+    )
+
+    mock_github_projects_instance = mocker.patch.object(
+        generator, "_LivingDocumentationGenerator__github_projects_instance"
+    )
+
+    repo_a = mocker.Mock()
+    repo_a.full_name = "OrgA/RepoA"
+    repo_b = mocker.Mock()
+    repo_b.full_name = "OrgA/RepoB"
+    generator._LivingDocumentationGenerator__github_instance.get_repo.side_effect = [repo_a, repo_b]
+
+    project_a = mocker.Mock(title="Project A")
+    project_b = mocker.Mock(title="Project B")
+    mock_github_projects_instance.get_repository_projects.side_effect = [[project_a], [project_b]]
+
+    project_status_1 = mocker.Mock()
+    project_status_1.status = "In Progress"
+
+    project_status_2 = mocker.Mock()
+    project_status_2.status = "Done"
+
+    project_issue_1 = mocker.Mock(spec=ProjectIssue)
+    project_issue_1.organization_name = "OrgA"
+    project_issue_1.repository_name = "RepoA"
+    project_issue_1.number = 1
+    project_issue_1.project_status = project_status_1
+
+    project_issue_2 = mocker.Mock(spec=ProjectIssue)
+    project_issue_2.organization_name = "OrgA"
+    project_issue_2.repository_name = "RepoA"
+    project_issue_2.number = 1
```

Comment:
This is the scenario, when we get from mining one project_issue twice, because it is connected to two projects. So we get two different project statuses. In this test, we check the case, when the issue_key is unique, if it is we create a list and append an issue into it. And if the project_issue with same key is already in our output, we just add the second project_state to the project_issue.

Reply:
Could you add some context comment to this test to explain it for future?

Reply:
Context comment added at: [28eec23](https://github.com/AbsaOSS/living-doc-generator/pull/67/commits/28eec2321bd0778e2b7c4609706286764145a26e).